collect_app/src/test/java/org/odk/collect/android/formentry/audit/FormSaveViewModelTest.java

```diff
@@ -368,6 +395,47 @@ public void resumeFormEntry_clearsSaveResult() {
         assertThat(saveResult.getValue(), equalTo(null));
     }
 
+    @Test
+    public void markOriginalFileOrDelete_whenFileNameIsEmpty_doesNotSaveFileForDelete() {
```

Comment:
It looks like all these new tests are mainly there to drive out the use of `SavedStateHandle`. I think a better way to test it would be more functionally by checking that if I call the methods that set state then use that saved state to create another view model then I get the behaviour I expect (probably that save or ignore do the right thing). This checks that the values are loaded from save state as well as written to it and also means we probably just need one test.